File: Front_Angular_Pathfinder/src/app/core/models/ConquistadorDTO.ts
import { ClaseDTO } from "./ClaseDTO";
import { EspecialidadDTO } from "./EspecialidadDTO";
import { SesionDTO } from "./SesionDTO";
import { UnidadDTO } from "./UnidadDTO";
import { UsuarioDTO } from "./UsuarioDTO";

export interface ConquistadorDTO {
    ConqId: number;
    ConqDni: string;
    ConqNombres: string;
    ConqApellidoPaterno: string;
    ConqApellidoMaterno: string;
    ConqFechaNacimiento: Date;
    ConqFechaInvestidura: Date;
    ConqCorreoPersonal: string;
    ConqCorreoCorporativo: string;
    ConqCelular: string;
    ConqTelefono: string;
    ConqAvance: string;
    ConqSexo: string;

    Usuario: UsuarioDTO;
    Sesion: SesionDTO;
    ConqClase: ClaseDTO;
    ConqUnidad: UnidadDTO;
    ConqEspecialidades: EspecialidadDTO[];
}

File: Back_Net8_Pathfinder2/Core/DTO/Conquistador/ConquistadorDTO.cs
using Core.Entities;

namespace Core.DTO;

public class ConquistadorDTO
{
    public int ConqId { get; set; }
    public string ConqDni { get; set; }
    public string ConqNombres { get; set; }
    public string ConqApellidoPaterno { get; set; }
    public string ConqApellidoMaterno { get; set; }
    public DateTime ConqFechaNacimiento { get; set; }
    public DateTime? ConqFechaInvestidura { get; set; }
    public string ConqCorreoPersonal { get; set; }
    public string? ConqCorreoCorporativo { get; set; }
    public string ConqCelular { get; set; }
    public string? ConqTelefono { get; set; }
    public string ConqAvance { get; set; }
    public string ConqSexo { get; set; }

    public UsuarioDTO Usuario { get; set; }
    public SesionDTO Sesion { get; set; }
    public ClaseDTO ConqClase { get; set; }
    public UnidadDTO ConqUnidad { get; set; }
    public ICollection<EspecialidadDTO> ConqEspecialidades { get; set; }

    public void CopyTo(ref Conquistador conquistador)
    {
        conquistador.PersId = ConqId;
        conquistador.PersDni = ConqDni;
        conquistador.PersNombres = ConqNombres;
        conquistador.PersApellidoPaterno = ConqApellidoPaterno;
        conquistador.PersApellidoMaterno = ConqApellidoMaterno;
        conquistador.PersFechaNacimiento = ConqFechaNacimiento;
        conquistador.ConqFechaInvestidura = ConqFechaInvestidura;
        conquistador.PersCorreoPersonal = ConqCorreoPersonal;
        conquistador.PersCorreoCorporativo = ConqCorreoCorporativo;
        conquistador.PersCelular = ConqCelular;
        conquistador.PersTelefono = ConqTelefono;
        conquistador.PersSexo = ConqSexo;
    }

    public void CopyFrom(ref Conquistador conquistador)
    {
        ConqId = conquistador.PersId;
        ConqDni = conquistador.PersDni;
        ConqNombres = conquistador.PersNombres;
        ConqApellidoPaterno = conquistador.PersApellidoPaterno;
        ConqApellidoMaterno = conquistador.PersApellidoMaterno;
        ConqFechaNacimiento = conquistador.PersFechaNacimiento;
        ConqFechaInvestidura = conquistador.ConqFechaInvestidura;
        ConqCorreoPersonal = conquistador.PersCorreoPersonal;
        ConqCorreoCorporativo = conquistador.PersCorreoCorporativo;
        ConqCelular = conquistador.PersCelular;
        ConqTelefono = conquistador.PersTelefono;
        ConqSexo = conquistador.PersSexo;
    }
}